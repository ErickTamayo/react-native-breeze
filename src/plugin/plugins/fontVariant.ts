import { PluginFunction } from "./types";

// prettier-ignore
export type PluginGroups = {
  variant: "small-caps" | "oldstyle-nums" | "lining-nums" | "tabular-nums" | "proportional-nums";
};

export const pattern = /^(?<variant>small-caps|oldstyle-nums|lining-nums|tabular-nums|proportional-nums)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { variant } = groups;
  return { fontVariant: [variant] };
};
