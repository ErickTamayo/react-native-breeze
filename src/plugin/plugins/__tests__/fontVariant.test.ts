import {
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";
import { pattern, plugin } from "../fontVariant";

describe("fontVariant", () => {
  // prettier-ignore
  const shouldMatch = ["small-caps", "oldstyle-nums", "lining-nums", "tabular-nums", "proportional-nums"];

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
