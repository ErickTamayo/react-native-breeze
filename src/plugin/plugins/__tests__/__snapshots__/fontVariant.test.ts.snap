// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fontVariant Should parse (lining-nums) correctly 1`] = `
Object {
  "fontVariant": Array [
    "lining-nums",
  ],
}
`;

exports[`fontVariant Should parse (oldstyle-nums) correctly 1`] = `
Object {
  "fontVariant": Array [
    "oldstyle-nums",
  ],
}
`;

exports[`fontVariant Should parse (proportional-nums) correctly 1`] = `
Object {
  "fontVariant": Array [
    "proportional-nums",
  ],
}
`;

exports[`fontVariant Should parse (small-caps) correctly 1`] = `
Object {
  "fontVariant": Array [
    "small-caps",
  ],
}
`;

exports[`fontVariant Should parse (tabular-nums) correctly 1`] = `
Object {
  "fontVariant": Array [
    "tabular-nums",
  ],
}
`;
